omegaconf/dictconfig.py

```diff
@@ -225,6 +229,21 @@ def is_typed(c: Any) -> bool:
             )
             raise ValidationError(msg)
 
+    def _check_assign_value_node(self, key: Any, value: Any) -> None:
+        from omegaconf import OmegaConf
+
+        element_type = self._metadata.element_type
+        value_type = OmegaConf.get_type(value)
+        if value_type is not Any and element_type != value_type:
```

Comment:
You are checking for type equality but the error message is talking about a subclass.
which one is it?

Reply:
Fix this.